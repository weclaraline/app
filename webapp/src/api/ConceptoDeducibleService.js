import ServiceAPI from "./ServiceAPI"

class ConceptoDeducibleService extends ServiceAPI {
    
    constructor(ConceptoDeducibleRequest){
        super();
        this.request = ConceptoDeducibleRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest()
                        .get("/recommendations",this.request);
        return data;
    }
}

export default ConceptoDeducibleService;