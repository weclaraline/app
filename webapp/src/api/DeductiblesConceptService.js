import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";
class DeductiblesConceptService extends ServiceAPI {
    
    constructor(ConceptoDeducibleRequest){
        super();
        this.request = ConceptoDeducibleRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .get("/recommendations",this.request);
        return data;
    }
}

export default DeductiblesConceptService;