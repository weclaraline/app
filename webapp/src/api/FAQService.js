import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";

class FAQService extends ServiceAPI {
    
    constructor(FAQRequest){
        super();
        this.request = FAQRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .get("/faq",this.request);
        return data;
    }
}

export default FAQService;