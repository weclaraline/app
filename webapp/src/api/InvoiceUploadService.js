import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";

class InvoiceUploadService extends ServiceAPI {
    
    constructor(InvoiceUploadRequest){
        super();
        this.request = InvoiceUploadRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .post("invoices/upload",this.request);
        return data;
    }
}

export default InvoiceUploadService;