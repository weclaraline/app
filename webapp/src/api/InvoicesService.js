import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";

class InvoicesService extends ServiceAPI {
    
    constructor(InvoicesRequest){
        super();
        this.request = InvoicesRequest;
    }

    async make(month, year){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .get(`invoices/${month}/${year}`,this.request);
        return data;
    }
}

export default InvoicesService;