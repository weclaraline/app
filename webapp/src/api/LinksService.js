import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";

class LinksService extends ServiceAPI {
    
    constructor(LinksRequest){
        super();
        this.request = LinksRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .get("/links",this.request);
        return data;
    }
}

export default LinksService;