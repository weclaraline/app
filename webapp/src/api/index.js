export { default as ServiceAPI } from './ServiceAPI'
export { default as StatusCodes } from './StatusCodes'
export { default as ConceptoDeducibleService } from './ConceptoDeducibleService'