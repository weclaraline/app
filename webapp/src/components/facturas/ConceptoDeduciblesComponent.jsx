import { Card, CardContent, Grid, Typography } from "@material-ui/core";
import React, { useEffect, useState } from "react";
import { ConceptoDeducibleService } from "../../api/";
import ConceptosDeduciblesListComponent from "../facturas/ConceptosDeducibles/ConceptosDeduciblesListComponent";

const conceptoDeducibleService = new ConceptoDeducibleService({});

const ConceptoDeduciblesComponent = () => {

    const [conceptosDeducibles, setConceptosDeducibles] = useState([]);
    
    const fetchConceptosDeducibles = async () => {
        const data = await conceptoDeducibleService.make();
        setConceptosDeducibles(data);
    }

    useEffect(() => {
        fetchConceptosDeducibles();
    }, []);
 
    return (<Card>
        <CardContent>
            <Grid item xs={12}>
                <Typography 
                    variant="h5" 
                    align="left" 
                    gutterBottom
                >
                    Conceptos
                </Typography>
            </Grid>
            <Grid item xs={12}>
                <ConceptosDeduciblesListComponent conceptosDeducibles={conceptosDeducibles}/>
            </Grid>
        </CardContent>
    </Card>);
}

export default ConceptoDeduciblesComponent;