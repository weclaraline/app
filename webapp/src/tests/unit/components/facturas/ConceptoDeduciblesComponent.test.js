import "../../testing_utils";
import { shallow } from "enzyme";
import ConceptoDeduciblesComponent from "../../../../components/facturas/ConceptoDeduciblesComponent";
import toJson from "enzyme-to-json";
import { render, unmountComponentAtNode } from "react-dom";
import { getByTestId, queryAllByTestId, queryByTestId, waitFor } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { ServiceAPI, StatusCodes } from "../../../../api/";
import nock from "nock";

let container = null;

const build = () => render(<ConceptoDeduciblesComponent />, container);

beforeEach(() => {
    nock.cleanAll();
    container = document.createElement("div");
    document.body.appendChild(container);
});
  
afterEach(() => {
    nock.cleanAll();
    unmountComponentAtNode(container);
    container.remove();
    container = null;
    return container;
});

it("Renders original content", () => {
    const tree = shallow(<ConceptoDeduciblesComponent />);
    expect(toJson(tree)).toMatchSnapshot();
});
