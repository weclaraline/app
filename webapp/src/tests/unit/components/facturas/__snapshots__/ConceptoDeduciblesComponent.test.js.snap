// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders original content 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        align="left"
        gutterBottom={true}
        variant="h5"
      >
        Conceptos
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <ConceptosDeduciblesListComponent
        conceptosDeducibles={Array []}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
